### PR TITLE
Fix input argument indexing error

### PR DESCRIPTION
**Pull request by doru1004**
This patch fixes an error with the indexing of un-initialized arguments when initialized arguments are present. There is no guarantee that the initialized arguments are at the beginning of the list of arguments.

This patch also adds error messages when unranked tensors are encountered during the shape inference pass.

A new flag was added to the onnx-mlir compiler to emit code before any pass (such as shape inference) is run over the initial ONNX Dialect code.

### CODE REVIEW

**Comment by tjingrant** [comment]
@doru1004 I'm curious what is your use case for EmitONNXBasic ?

File: src/Dialect/ONNX/ONNXOps.cpp [hunk 1]
```diff
@@ -514,7 +514,7 @@ bool ONNXAbsOp::inferShapes() {
 bool ONNXAddOp::inferShapes() {
   if (!getOperand(0).getType().isa<RankedTensorType>() ||
       !getOperand(1).getType().isa<RankedTensorType>()) {
-    emitError("ONNXAddOp inferShapes failed");
+    emitError("Shape tensor(s) not ranked");
```
Comment (tjingrant) [inline src/Dialect/ONNX/ONNXOps.cpp:517]: Not exactly understanding the error message - "Shape tensor(s) not ranked". The shape tensor (which is always 1D) is always ranked (with rank=1).

File: src/Builder/FrontendDialectTransformer.cpp [hunk 2]
```diff
@@ -434,10 +434,17 @@ class FrontendGenImpl {
     module_.push_back(entryPoint);
 
     // Map graph inputs to entry block arguments.
-    for (int i = 0; i < graph.input().size(); ++i)
+    // Counter of un-initialized tensors. This counter is used to index the
+    // entry block arguments.
+    int entryBlockIdx = 0;
```
Comment (tjingrant) [inline src/Builder/FrontendDialectTransformer.cpp:439]: `entryBlockIdx`  -> `entryBlockArgIdx`?

File: src/Dialect/ONNX/ONNXOps.cpp (hunk 1 above)
Comment (doru1004) [reply src/Dialect/ONNX/ONNXOps.cpp]: Fixed.

**Comment by doru1004** [comment]
> @doru1004 I'm curious what is your use case for EmitONNXBasic ?

This emits the ONNX dialect code before any pass is run. This is for inspecting the model source code that contains operations unsupported by the passes.

**Pull request merged**